### ISSUE TITLE
Update the bender API to look more like jbender

### ISSUE THREAD

**Issue by cgordon** [pinterest/bender#17]
The API for bender was created before the API for jbender (see https://github.com/pinterest/jbender). The API for jbender is more elegant and requires a LOT less memory allocations in the main loop (as it sends far fewer events). This would be a non-backwards compatible change, but given the way people vendor dependencies that should be alright for existing users.

**Comment by nuance** [comment]
The API should probably also be tweaked to use `time.Duration` and other non-numeric types where appropriate.

**Comment by cgordon** [comment]
Agreed, in fact, I made very poor use of the Go type system in all of this project, and it is due for a major overhaul.
